### PR TITLE
Re add support for Python2.7

### PR DESCRIPTION
**Pull request by mauguignard**

### CODE REVIEW

**Review by gonzalezzfelipe — APPROVED** [review]

File: pydruid/client.py [hunk 1]
```diff
@@ -564,7 +565,7 @@ def _post(self, query):
                     pass
 
             raise IOError(
-                "{0} \n Druid Error: {1} \n Query is: {2}".format(
+                "{0}\n Druid Error: {1}\n Query is: {2}".format(
```
Comment (klaussfreire) [inline pydruid/client.py]: Is this necessary?

File: pydruid/utils/query_utils.py [hunk 2]
```diff
@@ -28,9 +30,13 @@ def __init__(self, f, dialect="excel-tab", encoding="utf-8", **kwds):
...
         self.encoder = codecs.getincrementalencoder(encoding)()
 
-    @staticmethod
-    def __encode(data):
-        return str(data) if isinstance(data, str) else data
+    def __encode(self, data):
+        data = str(data) if isinstance(data, six.integer_types) else data
+        if not six.PY3:
+            data = data.encode("utf-8") if isinstance(data, unicode) else data  # noqa
+            data = data.decode("utf-8")
```
Comment (klaussfreire) [inline pydruid/utils/query_utils.py:37]: Why twice? Seems wrong.

File: requirements-dev.txt [hunk 3]
```diff
@@ -2,62 +2,46 @@
...
+pycurl==7.43.0.5
+pyparsing==2.4.7
+pytest==4.6.11
+pyyaml==5.4.1
+scandir==1.10.0
+singledispatch==3.6.1
+six==1.16.0
+toml==0.10.2
+tox==3.23.1
+typing==3.10.0.0 ; python_version < "3.0"
```
Comment (klaussfreire) [inline requirements-dev.txt:41]: Don't you need to specify the requirement alternatives for 3.x ?

File: setup.py [hunk 4]
```diff
@@ -45,9 +45,8 @@
     classifiers=[
         "License :: OSI Approved :: Apache Software License",
         "Programming Language :: Python",
+        "Programming Language :: Python :: 2.7",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
```
Comment (klaussfreire) [inline setup.py]: Why did you remove 3.6 and 3.8?

**Review by klaussfreire — CHANGES_REQUESTED** [review]

File: pydruid/utils/query_utils.py (hunk 2 above)
Comment (mauguignard) [reply pydruid/utils/query_utils.py:37]: It's doing an `encode` and then a `decode`. Why is that, I'm not sure. I copied it from the `0.5` branch: https://github.com/druid-io/pydruid/blob/0.5/pydruid/utils/query_utils.py#L35-L36 Since we are not going to use this (it's just meant for exporting `tsv` files and we are only going to use the DBAPI part), I didn't bother to see how it could be improved. In the `upstream` master branch this code does no longer exist either way.

File: requirements-dev.txt (hunk 3 above)
Comment (mauguignard) [reply requirements-dev.txt:41]: I tested this in both python2.7 and python3.7 environments and it's able to install all dependencies correctly as it is. We may bump some of these packages if we think they work better that way. But since these packages are only meant for develop and are not going to be installed on `pip install` I don't think it's worth it unless we have some issue with a particular version.

File: setup.py (hunk 4 above)
Comment (mauguignard) [reply setup.py]: TBH, I'm not sure. I'll restore them

File: pydruid/client.py (hunk 1 above)
Comment (mauguignard) [reply pydruid/client.py]: It generated trailing whitespace in the tests and flake8 didn't like it. I can leave it as it was and add a `noqa` it the test so it doesn't complain.

**Review by klaussfreire — APPROVED** [review]

File: pydruid/client.py (hunk 1 above)
Comment (mauguignard) [reply pydruid/client.py]: Ok, I was confused. It wasn't `flake8` but the `trailing-whitespace` hook from `pre-commit-hooks`. I think that hook doesn't have a way to skip a particular line :thinking: I'm not sure how it made it here since the `upstream` project also [asks to use it for contributing](https://github.com/druid-io/pydruid/#contributing). Since the hook automatically removes the trailing whitespace in the tests, it makes them fail if I don't change how the strings are created here.

Comment (mauguignard) [reply pydruid/client.py]: I'll create a PR to fix this in the `upstream` project as well.
